mmdet/apis/train.py

```diff
@@ -68,6 +68,54 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def scale_lr(cfg, distributed, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        distributed (bool): Distributed training or not.
+        logger (logging.Logger): logger.
+    """
+
+    if "mmdet_official_special_gpu_number" in cfg:
+        original_gpu_number = cfg.mmdet_official_special_gpu_number
+    else:
+        original_gpu_number = cfg.default_gpu_number
+
+    if "mmdet_official_special_samples_per_gpu" in cfg:
+        original_samples_per_gpu = cfg.mmdet_official_special_samples_per_gpu
+    else:
+        original_samples_per_gpu = cfg.default_samples_per_gpu
```

Comment:
It's maybe better to let users control the number of samples by themselves. The samples_per_gpu and learning rate are all can modified in config, but the number of GPU is modified in the command.

Reply:
User can control the number of samples by themselves by changing `data.samples_per_gpu` in the config file like before. This default_samples_per_gpu means the pretrain model default sample per gpu = 2 or mmdet_official_special_samples_per_gpu = 1 or 5 or 8 and so on

Reply:
User can control the number of samples by themselves by changing `data.samples_per_gpu` in the config file like before. This default_samples_per_gpu means the pretrain model default sample per gpu = 2 or mmdet_official_special_samples_per_gpu = 1 or 5 or 8 and so on

Reply:
Yes. My opinion is that we don't set default sample_per_gpu for users and users need to match the lr with their sample_per_gpu by themselves. We only scale the lr according to the number of GPU.

Reply:
It's easy to change the sample_per_gpu and lr for users in config. So, we left it to users. But for the GPU number, it can be changed in the command line. Thus, we need to automatically change it in codes.

Reply:
Oh, sorry i miss the point. This reason why i want to add the default sample per gpu is that the user may get OOM. I opened another PR #7377, that is based on this.